server-kotlin/src/main/kotlin/digital/cesko/internet_stream/InternetStreamService.kt

```diff
@@ -117,29 +118,36 @@ class InternetStreamService(
     fun parseBudgets(inputStream: InputStream): List<Budget> {
         val csvParser = CSVParser.parse(inputStream, Charsets.UTF_8, csvFormat)
         val records = csvParser.records
-        return records.map {
-            val type = it.get("DOKLAD_AGENDA")
-            val paragraph = it.get("PARAGRAF").toInt()
-            val item = it.get("POLOZKA").toInt()
-            val srcId = it.get("ORGANIZACE")
-            val name = it.get("ORGANIZACE_NAZEV")
-            val amountMd = it.get("CASTKA_MD").toBigDecimal()
-            val amountDal = it.get("CASTKA_DAL").toBigDecimal()
+        return records.mapNotNull {
+            createBudgetFromRecord(it)
+        }
+    }
+
+    fun createBudgetFromRecord(record: CSVRecord): Budget? {
+        val type = record.get("DOKLAD_AGENDA")
+        var budget: Budget? = null
+        if (type == "KDF" || type == "KOF") {
+            val paragraph = record.get("PARAGRAF").toInt()
+            val item = record.get("POLOZKA").toInt()
+            val srcId = record.get("ORGANIZACE")
+            val name = record.get("ORGANIZACE_NAZEV")
+            val amountMd = record.get("CASTKA_MD").toBigDecimal()
+            val amountDal = record.get("CASTKA_DAL").toBigDecimal()
             val amount = if (item < 5000) (amountMd - amountDal) else (amountDal - amountMd)
-            val event: Long? = it.get("ORGANIZACE").toLong()
-            val unit: Int? = it.get("ORJ").toInt()
-            val dateAsString: String? = it.get("DOKLAD_DATUM")
+            val event: Long? = record.get("ORGANIZACE").toLong()
+            val unit: Int? = record.get("ORJ").toInt()
+            val dateAsString: String? = record.get("DOKLAD_DATUM")
             val date: LocalDate? = if (dateAsString != null) LocalDate.parse(dateAsString) else null
-            val counterpartyId: String? = it.get("DOKLAD_DATUM")
-            val counterpartyName: String? = it.get("SUBJEKT_NAZEV")
-            val description: String? = it.get("POZNAMKA")
-            val year: Int? = it.get("DOKLAD_ROK").toInt()
-            val budget = Budget(
+            val counterpartyId: String? = record.get("DOKLAD_DATUM")
+            val counterpartyName: String? = record.get("SUBJEKT_NAZEV")
+            val description: String? = record.get("POZNAMKA")
+            val year: Int? = record.get("DOKLAD_ROK").toInt()
+            budget = Budget(
```

Comment:
(Optional) It might be more readable to not use the `val`s but insert the values directly to Bufget constructor using named parameters. Something like

```
Budget(
  paragraph = record.get("PARAGRAF").toInt(),
  ...
)
```